program-ui/settings/creators.tsx

```diff
@@ -0,0 +1,242 @@
+                    <FormControl
+                      isInvalid={
+                        !!getFieldState(`creators.${index}.address`, formState)
+                          .error
+                      }
+                    >
+                      <Input
+                        variant="filled"
+                        placeholder={PublicKey.default.toBase58()}
+                        {...register(`creators.${index}.address`)}
+                      />
+                      <FormErrorMessage>
+                        {
+                          getFieldState(`creators.${index}.address`, formState)
+                            .error?.message
+                        }
+                      </FormErrorMessage>
+                    </FormControl>
+                    <FormControl
+                      isInvalid={
+                        !!getFieldState(`creators.${index}.share`, formState)
+                          .error
+                      }
+                    >
+                      <InputGroup>
+                        <Input
+                          {...register(`creators.${index}.share`)}
+                          type="number"
+                        />
+                        <InputRightAddon children="%" />
+                      </InputGroup>
+                      <FormErrorMessage>
+                        {
+                          getFieldState(`creators.${index}.share`, formState)
+                            .error?.message
+                        }
+                      </FormErrorMessage>
+                    </FormControl>
+                    <IconButton
+                      borderRadius="md"
+                      isDisabled={
+                        watch("creators").length === 1 || formState.isSubmitting
+                      }
+                      colorScheme="red"
+                      icon={<IoMdRemove />}
+                      aria-label="remove row"
+                      onClick={() => remove(index)}
+                    />
+                  </Flex>
+                );
+              })}
+            </Flex>
+
+            {/* then render high level controls */}
+            <Flex>
+              <Button
+                leftIcon={<IoMdAdd />}
+                onClick={() => append({ address: "", share: 0 })}
+              >
+                Add Recipient
+              </Button>
+            </Flex>
+            <FormControl display="flex" as={Flex} flexDir="column">
+              <Flex gap={2} alignItems="center">
+                <Switch
+                  id="update-all"
+                  colorScheme="primary"
+                  {...register("updateAll")}
+                />
+                <FormLabel htmlFor="update-all" mb="0">
+                  Apply retroactively
+                </FormLabel>
+              </Flex>
+              <FormHelperText>
+                This will update the creators for all NFTs already minted on
+                this program.
```

Comment:
```suggestion
                Enabling this will update the creators for all NFTs already minted on
                this program. (This may be expensive depending on how many NFTs have already been minted.)
```